gc/gc-base/src/main/java/org/projectnessie/versioned/persist/gc/ContentBloomFilter.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.versioned.persist.gc;
+
+import static org.projectnessie.versioned.persist.gc.GCConfigConstants.CONF_NESSIE_GC_BLOOM_FILTER_FPP;
+import static org.projectnessie.versioned.persist.gc.GCConfigConstants.CONF_NESSIE_GC_BLOOM_FILTER_SIZE;
+
+import com.google.common.hash.BloomFilter;
+import com.google.common.hash.Funnels;
+import java.io.Serializable;
+import java.nio.charset.Charset;
+import java.util.Map;
+import org.projectnessie.model.Content;
+import org.projectnessie.model.IcebergTable;
+import org.projectnessie.model.IcebergView;
+
+/** A utility class wrapping bloom filter functionality. */
+public class ContentBloomFilter implements Serializable {
+
+  private final long expectedEntries;
+  private final double fpp;
+  private static final long serialVersionUID = -693336833916979221L;
+
+  // track iceberg table/view contents only using the snapshot/version id
+  // as the consumer of GC results needs only this info for clean up.
+  // String bloom filter with content type prefix + snapshot/version id.
+  private final BloomFilter<String> icebergContentBloomFilter;
+
+  private static final String ICEBERG_TABLE_PREFIX = "T";
```

Comment:
I want to save some characters here Instead of long strings from enum name as prefix is used for each content.

Reply:
Youi're not saving anything with that

Reply:
technically bloom filter has to create index based on few characters compared to using enum name (for each content). But I don't want to argue on this. I will change to Enum.name()

Reply:
If you care about that, then it would be better to have `Funnel` implementations per content-type.
But anyway - it's already way less than multiple bloom filters easily wasting megabytes of heap.